### PR TITLE
Update to boxo with refactored providerQueryManager.

### PR DESCRIPTION
**Pull request by hsanjuan**

### CODE REVIEW

**Comment by hsanjuan** [comment]
This is testing https://github.com/ipfs/boxo/pull/641

**Comment by hsanjuan** [comment]
@aschmahmann I have a problem: before, bitswap reprovided new blocks "immediately". Now, they are sent to the reprovider.System, which reprovides them when it likes, but not immediately.

The question is:
  * Whether I find a way to provide those immidiately like before (will have to port the queuing logic and call the underlying provider directly)
  * Whether I change failing tests and manually call reproviding so that they pass.
  
Opinions?

**Comment by aschmahmann** [comment]
@hsanjuan would you agree that this change in behavior seems more like a test issue than a production one? If so then what we do next seems like a matter of expediency vs maintenance.

For some of these tests it'd probably be nice to confirm that automatic providing works so we know if something breaks later. The reason this isn't working in tests is because this option https://github.com/ipfs/boxo/blob/c91cc1d5a0552c653936bd5051b6c0972d57743c/provider/reprovider.go#L226 (which is not exposed except for tests) defaults to waiting for 1 minute of uptime before providing to prevent wasted effort (e.g. if people run `ipfs add` but the daemon isn't running).

So there are some other options as well:
1. Remove the 1 minute timeout entirely (I'm pretty sure we wanted it for a reason, so may need to investigate more here on if removal is safe)
2. Modify how we do the tests to in some way allow setting that option (e.g. using the "Internal" section of the config, a test plugin with variables we can set in the config, etc.). It's not great that this would require building the test code / plugin into the main binary, but not necessarily so bad.

Not sure which option makes the most sense, part of me says that we should do whatever behavior we want to see normal users experience and compromise on if there's some grossness in the final binary / code related to testing (e.g. using the Internal config section or a test config section).

**Comment by hsanjuan** [comment]
After a quick discussion: what is happening is that the reprovider has an initialDelay and that's it. Otherwise it is going to provide blocks very quickly just like it happened below, except better. So the solution is to call Reprovide on the tests that need it by hand and move on.

File: test/cli/delegated_routing_v1_http_proxy_test.go [hunk 1]
```diff
@@ -60,8 +60,12 @@ func TestRoutingV1Proxy(t *testing.T) {
 		nodes := setupNodes(t)
 
 		cidStr := nodes[0].IPFSAddStr(testutils.RandomStr(1000))
-
-		res := nodes[1].IPFS("routing", "findprovs", cidStr)
+		// Reprovide as initialProviderDelay still ongoing
+		res := nodes[0].IPFS("bitswap", "reprovide")
+		if res.Err != nil {
+			t.Fatal(res.Err)
+		}
```
Comment (gammazero) [inline test/cli/delegated_routing_v1_http_proxy_test.go:67]: Minor: use `require` here and in other tests.
```suggestion
		require.NoError(t, res.Err)
```

**Review by gammazero — APPROVED** [review]

**Comment by hsanjuan** [comment]
@gammazero this should be ready too

**Comment by hsanjuan** [comment]
I have a flapping test at least. Not sure why but letting it sleep for a bit doesn't help. I need to look at it tomorrow.

**Review by gammazero — APPROVED** [review]

**Pull request merged**